### ISSUE TITLE
Elements should be able to opt-out of FocusStyleManager

### ISSUE THREAD

**Issue by adidahiya** [palantir/blueprint#5119]
#### Environment

- __Package version(s)__: core v3.53.0, v4.0.0-beta.13
- __Browser and OS versions__: Chrome, macOS

#### Feature request

Blueprint components should have a styling "escape hatch" to avoid the focus outline ring styling override applied by FocusStyleManager.

We often enable `FocusStyleManager.onlyShowFocusOnTabs()` in our applications, but sometimes there are use cases where I want to see focus rings in the UI when a tab interaction has not yet occurred.

This could be implemented very simply by adding an override class to `_focus-states.scss` and documenting its behavior:

```scss
// disable focus rings for everything in the UI, except places which have explicitly opted out of this behavior
.#{$ns}-focus-disabled :not(.#{$ns}-focus-style-override):focus {
  /* stylelint-disable declaration-no-important */
  outline: none !important;

  // ...
}
```

#### Examples

See this sandbox: https://codesandbox.io/s/walkthrough-popover-bj9q2?file=/src/CoreExample.css

Click the "projects" icon to open a "UI walkthrough" popover. Here, I have made the "Next" button autofocus because I want the user to be able to quickly move to the next step or exit the walkthrough via the keyboard. But since there are no focus rings, that doesn't appear to be a UI affordance.

![2022-02-08 17 30 26](https://user-images.githubusercontent.com/723999/153087748-2ea02383-5f56-495a-b25f-c95d4ad8508a.gif)

(in the sandbox, I have added hacky CSS overrides to demonstrate the desired functionality)
